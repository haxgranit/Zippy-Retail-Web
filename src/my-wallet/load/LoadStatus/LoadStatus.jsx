import { Button, Col, Row } from 'react-bootstrap';
import PageContainer from '../../../common/PageContainer';

export default function LoadStatus() {
  return (
    <PageContainer title="Personal Banking" subTitle="Made Fun With Zippy!">
      <div className="title">
        Load wallet
      </div>
      <div className="details">
        <Row>
          <Col xs={12}>Your $50 credited into your zippy account</Col>
        </Row>
        <Row>
          <Col xs={12}>
            Your total available Zippy balance is
            {' '}
            <b>$200</b>
          </Col>
        </Row>
      </div>

      <div className="action">
        <Button
          className="zippy-btn"
        >
          Close
        </Button>
      </div>
    </PageContainer>
  );
}
