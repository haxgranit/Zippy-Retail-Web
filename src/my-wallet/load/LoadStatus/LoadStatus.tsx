import PropTypes from 'prop-types';
import { Button, Col, Row } from 'react-bootstrap';
import { useLocation, useNavigate } from 'react-router-dom';
import { useEffect, useState } from 'react';
import { useMsal } from '@azure/msal-react';
import PageContainer from '../../../common/PageContainer';
import Api, { Account } from '../../../api';
import { useAppSelector } from '../../../app/hooks';
import { selectUser } from '../../../features/user/userSlice';

export default function LoadStatus({ mode } : { mode: string }) {
  const { instance, accounts } = useMsal();
  const navigate = useNavigate();
  const [state, setState] = useState<any>(useLocation().state);
  function retryPayment() {
    new Api(instance, accounts[0])
      .postFundingSourceTransaction(
        { amount: state.amount as number, isCredit: state.isCredit as boolean },
        state.sourceId,
      )
      .then(() => {
        setState({ ...state, status: 'completed' });
      })
      .catch((error) => {
        console.log('error', error);
      });
  }

  const [account, setAccount] = useState<Account | undefined>();
  const { user } = useAppSelector(selectUser);

  useEffect(() => {
    if (user) {
      new Api(instance, accounts[0])
        .listAccounts()
        .then((msalAccounts) => {
          setAccount(msalAccounts[0]);
        });
    }
  }, [user]);

  return (
    <PageContainer title="Your Wallet" subTitle="Made Fun With Zippy!">
      <div className="title">
        Fund Transfer
      </div>
      <div className="text-center">
        {state.status === 'pending' && (
          <i className="zippy-cash-icon zc-pending" />
        )}
        {state.status === 'completed' && (
          <i className="zippy-cash-icon zc-completed" />
        )}
        {state.status === 'failed' && (
          <i className="zippy-cash-icon zc-failed" />
        )}
      </div>
      <div className="details">
        {state.status === 'completed' && (
          <Row>
            <Col xs={12}>
              <p className="status-msg">
                Your $
                {state?.amount}
                {' '}
                credit into your zippy account
              </p>
            </Col>
          </Row>
        )}
        {state.status === 'pending' && (
          <Row>
            <Col
              xs={12}
            >
              <p className="status-msg">
                Your $
                {state?.amount}
                {' '}
                Zippy
                {' '}
                {mode === 'load' ? 'credited' : 'debit'}
                {' '}
                request is in process.

              </p>
            </Col>
          </Row>
        )}
        {state.status === 'failed' && (
        <Row>
          <Col
            xs={12}
          >
            <p className="status-msg">
              Your $
              {state?.amount}
              {' '}
              Zippy credit request
              has been failed.
            </p>
          </Col>
        </Row>
        )}
        {(state.status === 'completed' || state.status === 'pending') && (
          <Row>
            <Col xs={12} style={{ color: '#000000', padding: '0px 90px' }}>
              <p className="balance-msg">
                Your total available Zippy balance is
                {' '}
                <br />
                <b>
                  $
                  {account?.balance}
                  .00
                </b>
              </p>
            </Col>
          </Row>
        )}
      </div>

      <div className="action">
        {state.status !== 'failed' ? (
          <Button
            className="zippy-btn"
            onClick={() => navigate('/my-wallet/load/transfer-details', {
              state: { id: state.id, sourceId: state.sourceId, isCredit: state.isCredit },
            })}
          >
            View Transfer Detail
          </Button>
        ) : (
          <Button
            className="zippy-btn"
            onClick={() => retryPayment()}
          >
            Retry
          </Button>
        )}
      </div>
    </PageContainer>
  );
}

LoadStatus.propTypes = {
  mode: PropTypes.string,
};

LoadStatus.defaultProps = {
  mode: 'load',
};
