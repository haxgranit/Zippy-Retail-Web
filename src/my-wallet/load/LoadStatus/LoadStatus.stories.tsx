import { ComponentStory, ComponentMeta } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import LoadStatus from './LoadStatus';

export default {
  title: 'my-wallet/load/LoadStatus',
  component: LoadStatus,
} as ComponentMeta<typeof LoadStatus>;

const Template: ComponentStory<typeof LoadStatus> = () => (
  <BrowserRouter>
    <LoadStatus />
  </BrowserRouter>
);

export const Default = Template.bind({});
Default.args = {};
Default.parameters = {};
