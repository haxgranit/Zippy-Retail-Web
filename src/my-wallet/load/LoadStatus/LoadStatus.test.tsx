import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import LoadStatus from './LoadStatus';

const ReactTestRenderer = require('react-test-renderer');

const mockedUsedNavigate = jest.fn();
// const mockeduseLocation = jest.fn().mockImplementation(() => ({ state: { status: 'Success' } }));

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useParams: () => ({
    stepId: 'details',
  }),
  useLocation: () => ({ state: { status: 'Success' } }),
  useNavigate: () => mockedUsedNavigate,
}));

describe('TransactionComplete Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(
      <BrowserRouter>
        <LoadStatus />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
