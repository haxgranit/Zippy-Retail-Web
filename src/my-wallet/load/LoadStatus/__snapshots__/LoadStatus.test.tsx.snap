// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionComplete Component matches the snapshot 1`] = `
<div
  className=""
>
  <div
    className="zippy-cash-container-wrap"
  >
    <div
      className="backdrop "
    />
    <div
      className="zippy-cash-container"
    >
      <div
        className="zippy-cash-container-header"
      >
        <h1
          className="title"
        >
          Your Wallet
        </h1>
        <h2
          className="catch"
        >
          Made Fun With Zippy!
        </h2>
      </div>
      <div
        className="zippy-cash-container-body"
      >
        <div
          className="title"
        >
          Fund Transfer
        </div>
        <div
          className="text-center"
        />
        <div
          className="details"
        />
        <div
          className="action"
        >
          <button
            className="zippy-btn btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            View Transfer Detail
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
